registry/storage/driver/middleware/cloudfront/s3filter_test.go

```diff
@@ -0,0 +1,374 @@
+		},
+	})
+	defer server.Close()
+
+	ips := newAWSIPs(serverIPRanges(server), time.Hour, nil)
+	ips.tryUpdate()
+	AssertEqual(t, 1, len(ips.ipv4))
+}
+
+func TestInvalidNetworkType(t *testing.T) {
+	t.Parallel()
+	server := setupTest(aWSIPResponse{
+		Prefixes: []prefixEntry{
+			{IPV4Prefix: "192.168.0.0/24"},
+		},
+		V6Prefixes: []prefixEntry{
+			{IPV6Prefix: "ff00::/8"},
+			{IPV6Prefix: "fe00::/8"},
+		},
+	})
+	defer server.Close()
+
+	ips := newAWSIPs(serverIPRanges(server), time.Hour, nil)
+	AssertEqual(t, 0, len(ips.getCandidateNetworks(make([]byte, 17)))) // 17 bytes does not correspond to any net type
+	AssertEqual(t, 1, len(ips.getCandidateNetworks(make([]byte, 4))))  // netv4 networks
+	AssertEqual(t, 2, len(ips.getCandidateNetworks(make([]byte, 16)))) // netv6 networks
+}
+
+// RawMockHandler, like MockIPRangesHandler, sets
+// up a server to emulate aws. This handler
+// is configured by the json text, to facilitate testing
+// parsing.
+type RawMockHandler struct {
+	data string
+}
+
+func (m RawMockHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(m.data))
+}
+
+func TestParsing(t *testing.T) {
+	t.Parallel()
+	server := httptest.NewServer(RawMockHandler{
+		data: `{
+      "prefixes": [{
+        "ip_prefix": "192.168.0.0",
+        "region": "someregion",
+        "service": "s3"}],
+      "ipv6_prefixes": [{
+        "ipv6_prefix": "2001:4860:4860::8888",
+        "region": "anotherregion",
+        "service": "ec2"}]
+    }`,
+	})
+	defer server.Close()
+	schema, err := fetchAWSIPs(server.URL)
+
+	AssertEqual(t, nil, err)
+	AssertEqual(t, 1, len(schema.Prefixes))
+	AssertEqual(t, prefixEntry{
+		IPV4Prefix: "192.168.0.0",
+		Region:     "someregion",
+		Service:    "s3",
+	}, schema.Prefixes[0])
+	AssertEqual(t, 1, len(schema.V6Prefixes))
+	AssertEqual(t, prefixEntry{
+		IPV6Prefix: "2001:4860:4860::8888",
+		Region:     "anotherregion",
+		Service:    "ec2",
+	}, schema.V6Prefixes[0])
+}
+
+func TestUpdateCalledRegularly(t *testing.T) {
+	t.Parallel()
+
+	updateCount := 0
+	server := httptest.NewServer(http.HandlerFunc(
+		func(rw http.ResponseWriter, req *http.Request) {
+			updateCount++
+			rw.Write([]byte("ok"))
+		}))
+	defer server.Close()
+	newAWSIPs(fmt.Sprintf("%s/", server.URL), time.Second, nil)
+	time.Sleep(time.Second*4 + time.Millisecond*500)
+	if updateCount < 4 {
+		t.Errorf("Update should have been called at least 4 times, actual=%d", updateCount)
+	}
+}
+
+func TestClientCheck(t *testing.T) {
+	empty := context.TODO()
+
+	makeContext := func(useragent string) context.Context {
+		// makeContext returns a context with an http request with
+		// the given user agent
+		req := &http.Request{
+			Header: map[string][]string{
+				"User-Agent": {useragent},
+			},
+		}
+
+		return dcontext.WithRequest(empty, req)
+	}
+	cases := []struct {
+		Context  context.Context
+		Expected bool
+	}{
+		{Context: empty, Expected: false},
+		{Context: makeContext(""), Expected: false},
```

Comment:
No version number should default to http correct behavior, which means that no headers should be stripped server-side.